zoo/datacenters/amazon.py

```diff
@@ -0,0 +1,201 @@
+import itertools
+import pathlib
+
+import boto3
+from django.conf import settings
+from django.db import transaction
+
+from .models import InfraNode, NodeKind
+
+config_file = pathlib.Path(settings.AWS_CONFIG_FILE)
+credentials_file = pathlib.Path(settings.AWS_CREDENTIALS_FILE)
+
+if settings.AWS_CONFIG and not config_file.is_file():
+    config_file.parents[0].mkdir(parents=True, exist_ok=True)
+    config_file.write_text(settings.AWS_CONFIG)
+
+if settings.AWS_CREDENTIALS and not credentials_file.is_file():
+    credentials_file.parents[0].mkdir(parents=True, exist_ok=True)
+    credentials_file.write_text(settings.AWS_CREDENTIALS)
+
+
+def _get_url(value):
+    if value.startswith("dualstack."):
+        value = value[10:]
+    return value.strip().rstrip(".").replace("\\052", "*")
+
+
+def get_client(service_name, profile_name="default"):
+    return boto3.Session(profile_name=profile_name).client(service_name)
+
+
+def get_profiles(name="*"):
+    if "*" not in name:
+        return [name]
+
+    available = boto3.Session().available_profiles
+    selected = []
+
+    for profile in available:
+        prefix, suffix = name.split("*")
+        if profile.startswith(prefix) and profile.endswith(suffix):
+            selected.append(profile)
+
+    return selected
+
+
+def iter_hosted_zones(profile_name="default"):
+    route53 = get_client("route53", profile_name)
+    paginator = route53.get_paginator("list_hosted_zones")
+
+    for page in paginator.paginate():
+        yield from page["HostedZones"]
+
+
+def iter_resource_record_sets(hosted_zones_ids, profile_name="default"):
+    route53 = get_client("route53", profile_name)
+    for zone_id in hosted_zones_ids:
+        paginator = route53.get_paginator("list_resource_record_sets")
+
+        for page in paginator.paginate(HostedZoneId=zone_id):
+            yield from page["ResourceRecordSets"]
+
+
+def iter_load_balancers(names=None, profile_name="default"):
+    elb = get_client("elb", profile_name)
+    paginator = elb.get_paginator("describe_load_balancers")
+
+    if names:
+        page_iterator = paginator.paginate(LoadBalancerNames=names)
+    else:
+        page_iterator = paginator.paginate()
+
+    for page in page_iterator:
+        yield from page["LoadBalancerDescriptions"]
+
+
+def iter_load_balancers_v2(names=None, profile_name="default"):
```

Comment:
Couldn't this two be merged into one and passing the version as argument? Or it doesn't make sense?

Reply:
I think it makes sense. I will merge it.